source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -91,10 +90,32 @@ Config::SharedConfig::SharedConfig(
     max_downstream_connection_duration_ = std::chrono::milliseconds(connection_duration);
   }
 
-  if (config.has_access_log_flush_interval()) {
-    const uint64_t flush_interval =
-        DurationUtil::durationToMilliseconds(config.access_log_flush_interval());
-    access_log_flush_interval_ = std::chrono::milliseconds(flush_interval);
+  if (config.has_access_log_options()) {
+    if (config.flush_access_log_on_connected() /* deprecated */) {
+      throw EnvoyException("Only one of flush_access_log_on_connected from TcpProxy or"
+                           "access_log_options can be specified.");
+    }
```

Comment:
nit: not sure if `from TcpProxy` still necessary because both `flush_access_log_on_connected` and `access_log_options` from `TcpProxy`.